src/components/LocationErrorMessage/BaseLocationErrorMessage.js

```diff
@@ -0,0 +1,104 @@
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    locationErrorCode: undefined,
+};
+
+function BaseLocationErrorMessage({locationErrorCode, onAllowLocationLinkPress, translate}) {
+    if (!locationErrorCode) {
+        return null;
+    }
+
+    const isPermissionDenied = locationErrorCode === 1;
+
+    /**
+     * Clears the location error on press of close icon
+     */
+    const dismissError = () => {
+        User.clearLocationError();
+    };
+
+    return (
+        <View style={[styles.dotIndicatorMessage, styles.mt4]}>
+            <View style={styles.offlineFeedback.errorDot}>
+                <Icon
+                    src={Expensicons.DotIndicator}
+                    fill={colors.red}
+                />
+            </View>
+            <View style={styles.offlineFeedback.textContainer}>
+                {/* 
+                  Show appropriate error msg on location issues
+                  - errorCode = -1 -> location not supported (web only)
+                  - errorCode = 1 -> location permission is not enabled
+                  - errorCode = 2 -> location is unavailable or there is some connection issue
+                  - errorCode = 3 -> location fetch timeout  
+                */}
+                {isPermissionDenied ? (
+                    <Text>
+                        <Text style={[styles.offlineFeedback.text]}>{`${translate('location.permissionDenied')} ${translate('location.please')}`}</Text>
+                        <TextLink
+                            onPress={onAllowLocationLinkPress}
+                            style={[styles.offlineFeedback.text, styles.textBlue]}
+                        >
+                            {` ${translate('location.allowPermission')} `}
+                        </TextLink>
+                        <Text style={[styles.offlineFeedback.text]}>{translate('location.tryAgain')}</Text>
+                    </Text>
+                ) : (
+                    <Text style={styles.offlineFeedback.text}>{translate('location.notFound')}</Text>
+                )}
+            </View>
+            <View>
+                <Tooltip text={translate('common.close')}>
+                    <PressableWithoutFeedback
+                        onPress={dismissError}
+                        style={[styles.touchableButtonImage]}
+                        accessibilityRole={CONST.ACCESSIBILITY_ROLE.BUTTON}
+                        accessibilityLabel={translate('common.close')}
+                    >
+                        <Icon src={Expensicons.Close} />
+                    </PressableWithoutFeedback>
+                </Tooltip>
+            </View>
+        </View>
+    );
+}
+
+BaseLocationErrorMessage.displayName = 'BaseLocationErrorMessage';
+BaseLocationErrorMessage.propTypes = propTypes;
+BaseLocationErrorMessage.defaultProps = defaultProps;
+export default compose(
+    withOnyx({
+        locationErrorCode: {
+            key: ONYXKEYS.LOCATION_ERROR_CODE,
+        },
```

Comment:
I think this is a valid case where `initWithStoredValue: false` should be used. That way, when the component mounts, it would always start from a known state (assuming we always want to start with an empty error).

Reply:
I actually have the same question as @huzaifa-99. What is the reason to store this value in Onyx if we end up clearing the value every time?

Reply:
haha yes I asked exactly same question there 😅